### PR TITLE
validate input at forgot password page

### PR DESCRIPTION
**Pull request by juneluo**
### Description
**Fixes bug priority low 3 of LOGIN COMPONENT: Fix missing validation for email in the “forgot password” page**
a) The validation for email is missing in the “forgot password” page. Invalid email is accepted.
b) “Please enter your full name” appears if the submit button is clicked without first and last name. 
     But providing the values, doesn’t remove the error even after clicking submit.

### Mainly changes explained:
It will check whether the input matches the schema and show alert if not during the entering process and after clicking submit button.
img

### How to test:
check into current branch
do npm install and ... to run this PR locally
Login page --> forgot password link --> enter invalid email/ leave name empty -->enter valid email address and full name

### CODE REVIEW

**Review by Theophileaseh — APPROVED** [review]
## Approved

Hi @juneluo 

Great work. 🎉 
The app works perfectly fine and the validations work as expected. ✔️ 

https://user-images.githubusercontent.com/51954313/218248303-176c9c43-bc68-4275-97da-ea51279622d8.mp4

**Review by Theophileaseh — APPROVED** [review]
## Approved

Hi @juneluo 

Great work. 🎉 
The app works perfectly fine and the validations work as expected. ✔️ 

https://user-images.githubusercontent.com/51954313/218248303-176c9c43-bc68-4275-97da-ea51279622d8.mp4

**Review by Filipe-Oliv — APPROVED** [review]

**Review by Jorge-Nerd — COMMENTED** [review]
It seems to me that it is working very well

**Comment by Filipe-Oliv** [comment]
Hi @juneluo 

I tested your PR and it's working correctly.

[week 03 - 02-11-23 (#677).webm](https://user-images.githubusercontent.com/114119668/218263940-0ebfa2c6-f05e-4086-a9eb-a2a413f7e344.webm)

**Comment by AnishPandita** [comment]
Hello @juneluo I tested your PR #677 and It seems to be working great. I had a quick question even though there is a check to ensure the text boxes are not empty but the user can still put in blank spaces as names. Should there be a check that there is atleast one character entered?
Screenshot 2023-02-11 at 12 34 38 PM
Screenshot 2023-02-11 at 12 35 50 PM
Screenshot 2023-02-11 at 12 36 02 PM

**Review by RJBornToWin — CHANGES_REQUESTED** [review]
Good work here by using joi, combining multiple states into single user object.

I have 1 suggestion and 1 problem.
1. Problem: 
   a. Email without dot (.) is accepted

 Screen Shot 2023-02-11 at 6 11 43 PM

   b. Also on consequent changes, adding space bar to email adds "dot" (.)

https://user-images.githubusercontent.com/16258136/218289766-8494bcc2-7d05-4ff1-aaf7-90a96671a592.mov

   c. Instead of variable name being exposed or shown, please change it to descriptive words. 
       Eg: firstName to be changed as "First Name"

3. Suggestion:
Instead of validating on user entering the details, i suggest to check it on submit button click. 
Jae can take a call if this is okay or the behavior you have added is okay. @one-community  
https://user-images.githubusercontent.com/16258136/218289808-a74b9d3f-d123-4cc7-9823-4a827a7310d1.mov

**Comment by Gukal7** [comment]
Hi @juneluo, I have also tested your code and it works like required. But like @RJBornToWin noticed that the email is accepted without a dot (.). Also I believe the prompt should be like plain english "First Name" is not allowed to empty instead of like a coding variable "firstName".


Screenshot 2023-02-13 at 2 01 10 PM

**Comment by RafaelCastilhoCastro** [comment]
Hi @juneluo,
I tested your PR and your proposed changes are working as intended.
I tried to replicate the issue pointed out by @RJBornToWin where two spaces create a dot but it didn't happen with me.
Besides that, I observed the same results on other issues reported: email being accepted without ".sufix" and variables name on the alert messages.
Thanks

File: src/components/Login/ForgotPassword.jsx [hunk 1]
```diff
@@ -35,42 +60,85 @@ const ForgotPassword = React.memo(() => {
     }
   };
 
+  const handleInput = e => {
+    const { name, value } = e.target;
+    let errorData = { ...message };
+    //use schema.extract instead if joi package is updated
+    var vaildateResult = {};
```
Comment (vitorsaa2k) [inline src/components/Login/ForgotPassword.jsx:67]: there's a misspelling on the "validateResult" variable, which is being spelled as "vaildateResult"

**Review by tingedudu8 — COMMENTED** [review]
Hi @juneluo, I tested your PR and all the proposed changes mentioned above worked as intented. But still the email accepted special symbols that should not appear, like "&#!", please take a look, thanks!

https://user-images.githubusercontent.com/70128933/218597201-efdf33a2-7513-45a1-912d-f01e1ae03593.mp4

**Review by RJBornToWin — APPROVED** [review]
Works as intended.

**Comment by Filipe-Oliv** [comment]
First and last name popups are in inverted fields.

**Comment by one-community** [comment]
Wow, yes good catch Filipe! 

Please raise the PR fix soon and PM me with it @juneluo so I can just merge it without a bunch of reviews.

**Pull request merged**